editor.planx.uk/src/@planx/components/NumberInput/Public.test.tsx

```diff
@@ -84,6 +84,56 @@ test("allows negative numbers to be input when toggled on by editor", async () =
   expect(handleSubmit).toHaveBeenCalledWith({ data: { fahrenheit: -10 } });
 });
 
+test("a clear error is shown if decimal value added when onlyWholeNumbers is toggled on", async () => {
+  const handleSubmit = vi.fn();
+
+  const { user } = setup(
+    <NumberInput
+      fn="fahrenheit"
+      title="What's the temperature?"
+      handleSubmit={handleSubmit}
+      onlyWholeNumbers={true}
+    />,
+  );
+
+  expect(screen.getByRole("heading")).toHaveTextContent(
+    "What's the temperature?",
+  );
+
+  const textArea = screen.getByLabelText("What's the temperature?");
+
+  await user.type(textArea, "10.06");
+  await user.click(screen.getByTestId("continue-button"));
+
+  expect(screen.getByText("Enter a whole number")).toBeInTheDocument();
+  expect(handleSubmit).not.toHaveBeenCalledWith({
+    data: { fahrenheit: 10.06 },
+  });
```

Comment:
nit: `not.toHaveBeenCalled()` might be more meaningful & definitive check here? Because we want to check if the error _blocks_ submission, not that it was possibly automatically rounded to `fahrenheit: 10` and successfully submitted for example!